python/lsst/ip/isr/isrTask.py

```diff
@@ -1865,28 +1876,42 @@ def defineEffectivePtc(self, ptcDataset, detector, bfGains, overScans, metadata)
                     # Both serial and parallel overscan were
                     # run.  Only report noise from serial here.
                     noise = overscanResults.residualSigma[0]
+
+                # Overscan noise is always in adu; we must standardize
+                # the noise attributes of all PhotonTransferCurveDataset
+                # objects to units of electrons.
+                noise *= gain
             elif self.config.usePtcReadNoise:
                 # Try then with the PTC noise.
+                self.log.debug("Using noise from Photon Transfer Curve.")
                 noise = ptcDataset.noise[ampName]
                 noiseProvenanceString = "ptc"
-                self.log.debug("Using noise from Photon Transfer Curve.")
+
+                ptcVersion = ptcDataset.getMetadata()['PTC_VERSION']
+                if ptcVersion < 1.9 and ptcDataset.ptcFitType == "FULLCOVARIANCE":
+                    # Before version 1.9, the noise stored in the PTC was in
+                    # units of electron^2 only if ptcFitType == FULLCOVARIANCE.
+                    # After version 1.9, we standardized the
+                    # PhotonTransferCurveDataset noise units to electron to fix
+                    # this bug. If a user tries to use an earlier version of
+                    # PTC with this fit type, we must be sure to do the
+                    # calculations properly. More information about this noise
+                    # issue can be found in DM-45976.
+                    self.log.info(f"Input PTC VERSION ({ptcVersion}) < 1.9 and"
+                                  "ptcFitType == FULLCOVARIANCE. Applying fix"
+                                  f"for DM-45976 noise issue for AMP {ampName}.")
+                    noise = numpy.sqrt(noise)
             else:
                 # Finally, try with the amplifier noise.
-                # We already have a detector at this point. If there
-                # was no detector to begin with, one would have
-                # been created with self.config.gain and
-                # self.config.noise.
+                # We already have a detector at this point. If there was no
+                # detector to begin with, onewould have been created with
+                # self.config.gain and self.config.noise.
                 noise = amp.getReadNoise()
-
-            if math.isnan(gain):
-                gain = 1.0
-                self.log.warning("Gain for amp %s set to NAN! Updating to"
-                                 " 1.0 to generate Poisson variance.", ampName)
-            elif gain <= 0:
-                patchedGain = 1.0
-                self.log.warning("Gain for amp %s == %g <= 0; setting to %f.",
-                                 ampName, gain, patchedGain)
-                gain = patchedGain
+                # Amplifier object noise is always in adu; we must
```

Comment:
The read noise is always in `electron` in the amplifier object.  See https://github.com/lsst/afw/blob/7f102674c438d58292d6800526c800bf78397141/include/lsst/afw/cameraGeom/Amplifier.h#L120 and ticket description.

Reply:
Correct, I removed this application of gain